python/demo/demo_mixed-topology.py

```diff
@@ -0,0 +1,187 @@
+    basix.create_element(basix.ElementFamily.P, basix.CellType.prism, 1),
+]
+elements_cpp = [_cpp.fem.FiniteElement_float64(e._e, None, True) for e in elements]
+# NOTE: Both dofmaps have the same IndexMap, but different cell_dofs
+dofmaps = _cpp.fem.create_dofmaps(mesh.comm, mesh.topology, elements_cpp)
+
+# Create C++ function space
+V_cpp = _cpp.fem.FunctionSpace_float64(mesh, elements_cpp, dofmaps)
+
+# Create forms for each cell type.
+# FIXME This hack is required at the moment because UFL does not yet know about
+# mixed topology meshes.
+a = []
+for i, cell_name in enumerate(["hexahedron", "prism"]):
+    print(f"Creating form for {cell_name}")
+    element = basix.ufl.wrap_element(elements[i])
+    domain = ufl.Mesh(basix.ufl.element("Lagrange", cell_name, 1, shape=(3,)))
+    V = FunctionSpace(Mesh(mesh, domain), element, V_cpp)
+    u, v = ufl.TrialFunction(V), ufl.TestFunction(V)
+    k = 12.0
+    a += [(ufl.inner(ufl.grad(u), ufl.grad(v)) - k**2 * u * v) * ufl.dx]
+
+# Compile the form
+# FIXME: For the time being, since UFL doesn't understand mixed topology meshes,
+# we have to call mixed_topology_form instead of form.
+a_form = mixed_topology_form(a, dtype=np.float64)
+
+# Assemble the matrix
+A = assemble_matrix(a_form)
+
+# Solve
+A_scipy = A.to_scipy()
+b_scipy = np.ones(A_scipy.shape[1])
+
+x = spsolve(A_scipy, b_scipy)
+print(f"Solution vector norm {np.linalg.norm(x)}")
+
+# I/O
+# Save to XDMF
+xdmf = """<?xml version="1.0"?>
+<!DOCTYPE Xdmf SYSTEM "Xdmf.dtd" []>
+<Xdmf Version="3.0" xmlns:xi="https://www.w3.org/2001/XInclude">
+  <Domain>
+    <Grid Name="mesh" GridType="Collection" CollectionType="spatial">
+
+"""
+
+perm = [cell_perm_vtk(CellType.hexahedron, 8), cell_perm_vtk(CellType.prism, 6)]
+topologies = ["Hexahedron", "Wedge"]
+
+for j in range(2):
+    vtk_topology = []
+    geom_dm = mesh.geometry.dofmaps(j)
+    for c in geom_dm:
+        vtk_topology += list(c[perm[j]])
+    topology_type = topologies[j]
+
+    xdmf += f"""
+      <Grid Name="{topology_type}" GridType="Uniform">
+        <Topology TopologyType="{topology_type}">
+          <DataItem Dimensions="{geom_dm.shape[0]} {geom_dm.shape[1]}"
+           Precision="4" NumberType="Int" Format="XML">
+          {" ".join(str(val) for val in vtk_topology)}
+          </DataItem>
+        </Topology>
+        <Geometry GeometryType="XYZ" NumberType="float" Rank="2" Precision="8">
+          <DataItem Dimensions="{mesh.geometry.x.shape[0]} 3" Format="XML">
+            {" ".join(str(val) for val in mesh.geometry.x.flatten())}
+          </DataItem>
+        </Geometry>
+        <Attribute Name="u" Center="Node" NumberType="float" Precision="8">
+          <DataItem Dimensions="{len(x)}" Format="XML">
+            {" ".join(str(val) for val in x)}
+          </DataItem>
+       </Attribute>
+      </Grid>"""
+
+xdmf += """
+    </Grid>
+  </Domain>
+</Xdmf>
+"""
+
+fd = open("mixed-mesh.xdmf", "w")
+fd.write(xdmf)
+fd.close()
```

Comment:
This can be replaced with the new VTKHDF interface that @chris added in https://github.com/FEniCS/dolfinx/pull/3576

Reply:
Is there a way to write the solution using the stuff from this PR? I can only find `write_mesh`...